src/main/java/com/jaoafa/mymaid4/event/Event_AntiDisableInvisibilityWire.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * jaoLicense
+ *
+ * Copyright (c) 2021 jao Minecraft Server
+ *
+ * The following license applies to this project: jaoLicense
+ *
+ * Japanese: https://github.com/jaoafa/jao-Minecraft-Server/blob/master/jaoLICENSE.md
+ * English: https://github.com/jaoafa/jao-Minecraft-Server/blob/master/jaoLICENSE-en.md
+ */
+
+package com.jaoafa.mymaid4.event;
+
+import com.jaoafa.mymaid4.lib.EventPremise;
+import com.jaoafa.mymaid4.lib.MyMaidLibrary;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.EntityPotionEffectEvent;
+import org.bukkit.potion.PotionEffect;
+import org.bukkit.potion.PotionEffectType;
+
+public class Event_AntiDisableInvisibilityWire extends MyMaidLibrary implements Listener, EventPremise {
+    @Override
+    public String description() {
+        return "wireコマンドで出現させたコウモリの透明化エフェクトが切れた際、再度透明化エフェクトを付与します。";
+    }
+
+    @EventHandler
+    public void EntityDisableInvisibility(EntityPotionEffectEvent event) {
+        // エフェクトが発生したMOBのエンティティを取得する
+        Entity ent = event.getEntity();
+
+        if (ent.getType() != EntityType.BAT ||
+            event.getCause() != EntityPotionEffectEvent.Cause.valueOf("EXPIRATION") ||
+            event.getModifiedType() != PotionEffectType.getByName("INVISIBILITY") ||
+            !(ent.getScoreboardTags().contains("CmdWire1")) || !(ent.getScoreboardTags().contains("CmdWire2"))) {
```

Comment:
```suggestion
            !ent.getScoreboardTags().contains("CmdWire1") ||
            !ent.getScoreboardTags().contains("CmdWire2")) {
```

ifを改行する場合は条件をorまたはand毎に改行した方が分かりやすいです

Reply:
そもそもコードミスってたのでこれが正解です(コードは修正しました)
```java
event.getCause() != EntityPotionEffectEvent.Cause.EXPIRATION ||
            event.getModifiedType() != PotionEffectType.getByName("INVISIBILITY") ||
            !(ent.getScoreboardTags().contains("CmdWire1") || ent.getScoreboardTags().contains("CmdWire2")))
```

Reply:
```java
if (ent.getType() != EntityType.BAT ||
    event.getCause() != EntityPotionEffectEvent.Cause.EXPIRATION ||
    event.getModifiedType() != PotionEffectType.getByName("INVISIBILITY") ||
    !ent.getScoreboardTags().contains("CmdWire1") || 
    !ent.getScoreboardTags().contains("CmdWire2"))
```
にしない理由は…？別にかまわんちゃかまわんけど

Reply:
「CmdWire1またはCmdWire2のどちらかのタグが付いている」の否定
→「CmdWire1,CmdWire2のどちらのタグも付いていない」という状況が正しい処理なので多分このままにするしか無いと思われます
やり方あったら教えて下さい

Reply:
それだと「CmdWire1のタグが付いていない または CmdWire2のタグが付いていない」場合なので
「CmdWire1,CmdWire2のタグがどちらも付いている」場合じゃないと再透明化を実行してくれんので動かない
CmdWire1はリードを付けられている側、CmdWire2はリードを持っている側に付くタグ

Reply:
だったら

```java
if (ent.getType() != EntityType.BAT ||
    event.getCause() != EntityPotionEffectEvent.Cause.EXPIRATION ||
    event.getModifiedType() != PotionEffectType.getByName("INVISIBILITY") ||
    !(ent.getScoreboardTags().contains("CmdWire1") && ent.getScoreboardTags().contains("CmdWire2")))
```
では？
ORだと私が示したやつと変わらんと思うんだけど。若干あたまこんがらがってるからあれ

Reply:
数学わからん

Reply:
一つのifに全部まとめること自体の可読性懸念はあるよね

Reply:
```java
// 簡略化のため{}なし
if(ent.getType() != EntityType.BAT) return; // コウモリではない
if(event.getCause() != EntityPotionEffectEvent.Cause.EXPIRATION) return; // 有効期限切れ原因
if(!event.getModifiedType().equals(PotionEffectType.INVISIBILITY)) return; // 透明ではない
if(!ent.getScoreboardTags().contains("CmdWire1") && !ent.getScoreboardTags().contains("CmdWire2")) return; // CmdWire1 も CmdWire2 もついてない
```

Reply:
とりあえず上を書き方を参考にして変更しました